src/random.h

```diff
@@ -215,6 +215,7 @@ class RandomMixin
             // number becomes the new bitbuf.
             uint64_t gen = Impl().rand64();
             ret = (gen << bitbuf_size) | bitbuf;
+            assert(bits >= bitbuf_size);
```

Comment:
I tried to track down if/how it could be a race condition, but given that it's in init and all locks seems to be in place, I don't see how that could possibly be the case.